base/compiler/abstractinterpretation.jl

```diff
@@ -787,13 +787,13 @@ function abstract_call_builtin(interp::AbstractInterpreter, f::Builtin, fargs::U
         cnd = argtypes[2]
         if isa(cnd, Conditional)
             newcnd = widenconditional(cnd)
+            tx = argtypes[3]
+            ty = argtypes[4]
             if isa(newcnd, Const)
                 # if `cnd` is constant, we should just respect its constantness to keep inference accuracy
-                return newcnd.val ? tx : ty
+                return newcnd.val::Bool ? tx : ty
```

Comment:
This is implicit in `?`, so we don't typically annotate them

Reply:
Ah, I didn't know that. Maybe we add `typeassert` in optimization phase ?